outlines/fsm/guide.py

```diff
@@ -147,14 +117,14 @@ def get_next_instruction(self, state: CFGState) -> Instruction:
         """
 
         if state.parser_state is None:
-            return Write(torch.tensor([self.eos_token_id]))
+            return self.eos_token_id
 
         valid_tokens = list(
             self.iter_valid_token_ids(state, self.tokenizer.vocabulary.values())
         )
         if len(valid_tokens) == 1:
-            return Write(torch.tensor(valid_tokens))
-        return Generate(torch.tensor(valid_tokens))
+            return valid_tokens[0]
+        return None
```

Comment:
These are examples of broken instructions now, I think we need to harshly limit changes and do not propagate it further in the code's logic, by asking for `get_next_instruction` consumers to adjust to new types of instructions. If it expects a tensor here, it should get one. And we need to try to limit the scope to the `RegexGuide`. At least until we see what's what. We might create a plan for improving Guide's logics in following PRs once we confirm that it could be.